frontend/README.md

```diff
@@ -0,0 +1,389 @@
+
+**Pros:**
+
+- You can keep your app lightweight, as you avoid adding extra libraries to the build.
+- Reducers are nativaly supported by React and can be used for scoped portions of the app or globally depending on the use case.
+- Providers, context and hooks are great tools to spread the data across the component tree.
+
+**Cons:**
+
+- React still is a UI library, which doesn't include anything for I/O operations.
+- Everything is added at the lifecycle level, so functions, so handling caching across the data, functions and other data types needs to be manually handled by the developers.
+- As great as hooks can be, it can become cumbersome to have all of your business logic encapsulated in a concept that is framework specific.
+
+### Redux-Toolkit
+
+This is a great tool that works as a wrapper for the Redux library, it includes simpler APIs to enable fast development while still using Redux.
+**Pros:**
+
+- Includes simple ways for I/O implementations, and auto generates hooks for you to use, it also includes caching, error handling and more.
+- You can keep using the same patterns as for Redux with Actions, Reducers if you want to keep having fine control of the state.
+- Fully typed, so you can use it alongside `Typescript`.
+- Side effects come out of the box by allowing cross action listeners for reducers.
+
+**Cons:**
+
+- It comes with an initial overhead of learning how Redux works, as it feels targeted for old users that want a new way of interacting with he same pattern.
+- Includes several tools and APIs that you wont need unless you have a specific use case, it might be considered overkill for some apps.
+- It includes its own custom way when creating I/O operations, in case you have generic logic you want to reuse you'll have to migrate them.
+
+### React-Query
+
+React query is an I/O abstraction that allows you create custom logic to connect to an API while also provides its own state management.
+
+**Pros:**
+
+- Allows you to decouple the actual API call from the React code, having your generic gateways doing the job.
+- Native support for queries, mutations, paginated queries and more.
+- Includes a tagging system for cache, so you can decide when to refetch certain queries.
+
+**Cons:**
+
+- It doesn't include a custom state management.
+- It still uses hooks a primary API.
+
+### Decision
+
+Having worked on big enterprise level React apps and at the same time on smaller code bases, there are some compromises that you need to do based on the team, state of the business and overall development experience.
+
+I believe that having overkill startegies when is not necessary can be as harmful as not having them when its necessary as it can slow the development process by adding extra overhead or quirks just to support something that can be simply done by just using React.
```

Comment:
```suggestion
I believe that having overkill startegies when not necessary can be as harmful as not having them when it is necessary as it can slow the development process by adding extra overhead or quirks just to support something that can be simply done by just using React.
```